### PR TITLE
TDD-RUBY

### PR DESCRIPTION
**Pull request by AbuTalha3**
In this part of the project we did the following:
- [x] Wrote unit tests using Rspec.
- [x] Created a class called `Solver`.
- [x] Created a method called `factorial` that takes one argument, an integer `N`, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number. The factorial is the multiplication of all integers from 1 to `N` and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.
- [x] Created a method called `reverse` that takes one argument, a string `word`, and returns `word` reversed (e.g. if `word` is `"hello"` it returns `"olleh"`).
- [x] Created a method called `fizzbuzz` that takes one argument, an integer `N`, and returns a string. The returned string is constructed following these rules:
  - When `N` is divisible by 3, return `"fizz"`.
  - When `N` is divisible by 5, return `"buzz"`.
  - When `N` is divisible by 3 and 5, return `"fizzbuzz"`.
  - Any other case, return `N` as a string (e.g. say `N` is `7` then return `"7"`).

### CODE REVIEW

File: .github/workflows/linters.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ jobs:
   rubocop:
     name: Rubocop
     runs-on: ubuntu-22.04
-
+
```
Comment (HAMMAS-SALEEM) [inline .github/workflows/linters.yml:9]: image
image

- [ ] It would be nice if you fix the highlighted link in the images above 👀 to make your readme descriptive and professional 👍

Comment (HAMMAS-SALEEM) [inline .github/workflows/linters.yml:9]: **[OPTIONAL BUT HIGHLY RECOMMENDED]**

- [ ] It would be nice if you make your own logo of your name and put that in the readme file at the top to present yourself as a brand in the job market or tech industry to catch the recruiters' eyes 👍

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
Ukhv

## Hey @AbuTalha3, @lucy-sees :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements
✔️ Great work on follow the TDD approach 👍
✔️ Great work on making the tests concise 🔥
✔️ Great work on testing the edge cases 😮

Good job on implementing some of the requirements, There are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 2]
```diff
@@ -2,15 +2,15 @@
...
+  <h1><b>TDD-RUBY-SPEC PROJECT</b></h1>
 
 </div>
 
 <!-- TABLE OF CONTENTS -->
 
 ## 📗 Table of Contents
 
-- [📖 TDD ](#-about-project-)
+- [📖 TDD-RUBY-SPEC PROJECT ](#-about-project-)
```
Comment (OyePriscilla) [inline README.md]: - [ ] As mentioned earlier, kindly ensure that this link address is the same as the name given to the link element in line 31 of your README file. The link does not redirect as of the moment. 🙏 

## Sample

Untitled

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @AbuTalha3  and @lucy-sees 👏 👏 👏 👏, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

so-close-so-close-seth-meyers

 ## Highlights
Issue page redirected to the correct page ✔️ 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by skyv26 — APPROVED** [review]
## Hi @AbuTalha3 and @lucy-sees , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## STATUS: APPROVED 🟢 



## Highlights
- Implemented the features ✅ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
